File: guides/micronaut-security-session-database-authentication/java/src/main/java/example/micronaut/AuthoritiesFetcherService.java
package example.micronaut;

import example.micronaut.repositories.UserRoleJdbcRepository;
import jakarta.inject.Singleton;

import java.util.List;

@Singleton // <1>
class AuthoritiesFetcherService implements AuthoritiesFetcher {

    private final UserRoleJdbcRepository userRoleJdbcRepository;

    AuthoritiesFetcherService(UserRoleJdbcRepository userRoleJdbcRepository) { // <2>
        this.userRoleJdbcRepository = userRoleJdbcRepository;
    }

    @Override
    public List<String> findAuthoritiesByUsername(String username) {
        return userRoleJdbcRepository.findAllAuthoritiesByUsername(username);
    }
}

File: guides/micronaut-database-authentication-provider/python/src/example/micronaut/authorities_fetcher_service.py
from jakarta.inject import Singleton

from .authorities_fetcher import AuthoritiesFetcher
from .user_role_jdbc_repository import UserRoleJdbcRepository


@Singleton  # <1>
class AuthoritiesFetcherService(AuthoritiesFetcher):
    def __init__(self, user_role_jdbc_repository: UserRoleJdbcRepository):  # <2>
        self.user_role_jdbc_repository = user_role_jdbc_repository

    def findAuthoritiesByUsername(self, username: str) -> list[str]:
        return self.user_role_jdbc_repository.findAllAuthoritiesByUsername(username)
